src/user.rs

```diff
@@ -0,0 +1,46 @@
+use std::fs::File;
+use std::io::{Read, Write};
+use std::path::{Path, PathBuf};
+
+use rustc_serialize::json;
+
+use project_types::Project;
+use Error;
+
+
+pub fn new_user<P: AsRef<Path>>(project: String, public: P, name: String) -> Result<(), Error> {
```

Comment:
More relevantly, a side effect is something like writing a file to disk, changing the working directory, using up entropy in /dev/random, etc.

Depending on the state of the world means reading from disk, getting user input, accessing a web page, etc.